core/metamod.cpp

```diff
@@ -417,11 +417,17 @@ FileSystemFactory(const char *iface, int *ret)
 void
 mm_LogMessage(const char *msg, ...)
 {
+	static bool g_logging = false;
+	if (g_logging) {
+		return;
+	}
+	g_logging = true;
```

Comment:
This change exist, because I happened to be log looping to hell while debugging sourcehook.

Reply:
Use RAII. (I think MM:S doesn't use AMTL, but if it does, there's a helper for this pattern called SaveAndSet).